jupyterlab_git/git.py

```diff
@@ -976,11 +983,15 @@ async def pull(self, path, auth=None, cancel_on_conflict=False):
 
         return response
 
-    async def push(self, remote, branch, path, auth=None, set_upstream=False):
+    async def push(
+        self, remote, branch, path, auth=None, set_upstream=False, force=False
+    ):
         """
         Execute `git push $UPSTREAM $BRANCH`. The choice of upstream and branch is up to the caller.
         """
         command = ["git", "push"]
+        if force:
+            command.append("--force")
```

Comment:
We should be less aggressive by using `--force-with-lease`

```suggestion
            command.append("--force-with-lease")
```

See documentation: https://git-scm.com/docs/git-push